src/main/java/org/opensearch/neuralsearch/sparse/NeuralSparseIndexShard.java

```diff
@@ -0,0 +1,268 @@
+    @SneakyThrows
+    public void clearCache() {
+        try (Engine.Searcher searcher = indexShard.acquireSearcher(clearCacheSearcherSource)) {
+            for (final LeafReaderContext leafReaderContext : searcher.getIndexReader().leaves()) {
+                final LeafReader leafReader = leafReaderContext.reader();
+
+                // Find all sparse token fields in this segment
+                final Set<FieldInfo> sparseFieldInfos = collectSparseFieldInfos(leafReader);
+
+                // Get segment info for creating cache keys
+                final SegmentReader segmentReader = Lucene.segmentReader(leafReader);
+                final SegmentInfo segmentInfo = segmentReader.getSegmentInfo().info;
+
+                // Clear in-memory cache for each sparse field
+                for (FieldInfo fieldInfo : sparseFieldInfos) {
+                    if (!PredicateUtils.shouldRunSeisPredicate.test(segmentInfo, fieldInfo)) {
+                        continue;
+                    }
+                    CacheKey cacheKey = new CacheKey(segmentInfo, fieldInfo);
+                    ClusteredPostingCache.getInstance().removeIndex(cacheKey);
+                    ForwardIndexCache.getInstance().removeIndex(cacheKey);
+                }
+            }
+        } catch (Exception e) {
+            log.error("[Neural Sparse] Failed to acquire searcher", e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Warm up using CacheGated readers that automatically populate cache on read
+     */
+    private void warmUpWithCacheGatedReaders(LeafReader leafReader, FieldInfo fieldInfo, CacheKey key) throws IOException {
+        final SegmentReader segmentReader = Lucene.segmentReader(leafReader);
+        final SegmentInfo segmentInfo = segmentReader.getSegmentInfo().info;
+        final int docCount = segmentInfo.maxDoc();
+        final BinaryDocValues binaryDocValues = leafReader.getBinaryDocValues(fieldInfo.name);
+        if (binaryDocValues == null) {
+            log.error("[Neural Sparse] No binary doc values found for field: {}", fieldInfo.name);
+            return;
+        }
+        try {
+            // Create SparseTermsLuceneReader for inverted index
+            final SparseTermsLuceneReader luceneReader = new SparseTermsLuceneReader(createSegmentReadState(segmentInfo));
+
+            final CacheGatedPostingsReader postingsReader = new CacheGatedPostingsReader(
+                fieldInfo.name,
+                ClusteredPostingCache.getInstance().getOrCreate(key).getReader(),
+                ClusteredPostingCache.getInstance().getOrCreate(key).getWriter((ramBytesUsed) -> {
+                    throw new CircuitBreakingException("Circuit Breaker reaches limit", CircuitBreaker.Durability.PERMANENT);
+                }),
+                luceneReader
+            );
+            warmUpInvertedIndex(postingsReader);
+        } catch (Exception e) {
+            if (e instanceof CircuitBreakingException) {
```

Comment:
same, no need to catch from internal logic if you already catch it from external